images/prow-tests/Dockerfile

```diff
@@ -12,42 +12,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM gcr.io/k8s-testimages/kubekins-e2e:v20210113-cc576af-master AS stable-stuff
-
-ENV DEBIAN_FRONTEND noninteractive
+# Starting from the gcloud debian image https://github.com/GoogleCloudPlatform/cloud-sdk-docker/blob/master/debian_slim/Dockerfile
 ARG GCLOUD_VERSION=368.0.0
-ARG KUBECTL_VERSION=v1.21.4
 
-# If we don't run update, the apt-get install below don't work
-RUN apt-get update
+FROM google/cloud-sdk:${GCLOUD_VERSION}-slim AS stable-stuff
 
-# gcloud
-RUN gcloud components update --version=${GCLOUD_VERSION}
+ENV DEBIAN_FRONTEND noninteractive
+
+ARG KUBECTL_VERSION=v1.21.4
+ARG BAZELISK_VERSION=v1.11.0
+ARG DOCKER_VERSION=5:20.10.9~3-0~debian-buster
 
 # kubectl
-RUN rm -f "$(which kubectl)" && \
-    wget "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" -O /usr/local/bin/kubectl && \
+RUN curl -fsSL "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" -o /usr/local/bin/kubectl && \
     chmod +x /usr/local/bin/kubectl
 
-# Extra tools through apt-get
-RUN apt-get install -y uuid-runtime   # for uuidgen
-RUN apt-get install -y shellcheck     # for shellcheck presubmit
+# bazel
+RUN curl -fsSL "https://github.com/bazelbuild/bazelisk/releases/download/${BAZELISK_VERSION}/bazelisk-linux-amd64" -o /usr/local/bin/bazel && \
+    chmod +x /usr/local/bin/bazel
+
+# Docker including docker-ce, docker-ce-cli, and containerd.io
```

Comment:
We should need other Docker setups in https://github.com/kubernetes/test-infra/blob/master/images/bootstrap/Dockerfile#L79-L126 as well.